script.js

```diff
@@ -1,4 +1,322 @@
+  var ace = findAceIn(playerHand);
+  console.log(ace);
+
+  var playerPoints = sumCardPointsIn(playerHand);
+  console.log(sumCardPointsIn(playerHand));
+  if (playerPoints < 11 && ace == true) {
+    for (var i = 0; i < playerHand.length; i++) {
+      currentCard = playerHand[i];
+      var currentCardName = currentCard.name;
+      if (currentCardName == "ace") {
+        currentCard.points = 11;
+      }
+    }
+    playerPoints = sumCardPointsIn(playerHand);
+  }
+  var computerPoints = sumCardPointsIn(dealerHand);
+  console.log("computer points " + computerPoints);
+  console.log("player points " + playerPoints);
+  if (
+    (playerPoints > computerPoints && playerPoints <= 21) ||
+    (playerPoints < computerPoints && computerPoints > 21)
+  ) {
+    return `You 🌸 🌼 W I N 🌼 🌸 <br>${playerName}, you've got ${playerPoints} points. <br>Player's hand: ${printCardsIn(
+      playerHand
+    )} <br> Computer has ${computerPoints} points. <br>Computer's hand: ${printCardsIn(
+      dealerHand
+    )} <br> Play again? (yes/no)`;
+  } else if (
+    (playerPoints < computerPoints && computerPoints <= 21) ||
+    (playerPoints > computerPoints && playerPoints > 21)
+  ) {
+    return `You  😭 L O S T 😭 <br>Computer wins with ${computerPoints} points. <br>Computer's hand: ${printCardsIn(
+      dealerHand
+    )} <br> ${playerName}, you have ${playerPoints} points. <br>Player's hand: ${printCardsIn(
+      playerHand
+    )} <br> Play again? (yes/no)`;
+  } else if (playerPoints == computerPoints && playerPoints <= 21) {
+    return `It's a draw.<br>Player's hand: ${printCardsIn(
+      playerHand
+    )}<br>Computer's hand: ${printCardsIn(
+      dealerHand
+    )}<br> Play again? (yes/no)`;
+  }
+}
+// mode: play again
+function refreshRound() {
+  playerHand = [];
+  dealerHand = [];
+  deck = makeDeck();
+  shuffledDeck = shuffleCards(deck);
+  mode = "deal cards";
+  return `Let's play!`;
+}
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (mode == "welcome") {
+    var output = askForName();
+    mode = "greet with rules";
+    return output;
+  } else if (mode == "greet with rules") {
+    output = greetWithRules(input);
+    mode = "deal cards";
+    return output;
+  } else if (mode == "deal cards") {
+    var output = dealCards();
+    mode = "hit or stand";
+    return output;
+  } else if (mode == "hit or stand") {
+    if (input == "hit") {
+      output = hit();
+      return output;
+    } else if (input == "stand") {
+      output = hitDealer();
+      mode = "compute results";
+      return output;
+    }
+  } else if (mode == "compute results") {
+    output = computeResults();
+    mode = "play again?";
+    return output;
+  } else if (mode == "play again?") {
+    if (input == "yes") {
+      output = refreshRound();
+      mode = "deal cards";
+    } else if (input == "no") {
+      output = `Bye bye! Let's play again soon ~`;
+    }
+    return output;
```

Comment:
Extremely good effort with this. Can tell you did some refactoring of your code. The main function is super clean and very readable. Even if someone did not know how to play black jack, your code makes it easy to understand!